full_storage_utilization_test.py

```diff
@@ -0,0 +1,195 @@
+import time
+from sdcm.tester import ClusterTester
+
+
+class FullStorageUtilizationTest(ClusterTester):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.soft_limit = 75
+        self.hard_limit = 90
+        self.stress_cmd_30mins = self.params.get('stress_cmd')
+        self.add_node_cnt = self.params.get('add_node_cnt')
+        self.action_type = self.params.get('action_type')
+
+    def prepare_dataset_layout(self, dataset_size, row_size=10240):
+        n = dataset_size * 1024 * 1024 * 1024 // row_size
+        seq_end = n * 100
+
+        return f'cassandra-stress write cl=ONE n={n} -mode cql3 native -rate threads=10 -pop dist="uniform(1..{seq_end})" ' \
+               f'-col "size=FIXED({row_size}) n=FIXED(1)" -schema "replication(strategy=NetworkTopologyStrategy,replication_factor=3)"'
+
+    def setUp(self):
+        super().setUp()
+        self.start_time = time.time()
+
+    def start_slow_write(self):
+        self.run_stress_thread(stress_cmd=self.stress_cmd_30mins, keyspace_name=f"{self.action_type}")
+        self.log.info("Wait for 2 mins for the stress command to start")
+        time.sleep(120)        
+        
+    def scale_out(self):
+        self.start_slow_write()
+        self.log.info("Adding a new node")
+        self.add_new_node()
+        time.sleep(1800)
+
+    def scale_in(self):        
+        self.start_slow_write()
+        self.log.info("Removing a node")
+        self.remove_node()
+        time.sleep(1800)
+
+    def drop_data(self, keyspace_name):
+        '''
+        Drop keyspace and clear snapshots.
+        '''
+        node = self.db_cluster.nodes[0]
+        self.log.info("Dropping some data")
+        cql = f"DROP KEYSPACE {keyspace_name}"
```

Comment:
if decreasing dataset by dropping a keyspace, there could be no background read for it, if it is relevant somehow.

Reply:
thanks, I didn't know that about drop keyspace. I think in this case, we don't read/write into that keyspace (keyspace_large1) - its mostly idle after disk usage across first 20%